pkg/aws/client.go

```diff
@@ -33,20 +40,95 @@ type ManifestRetriever interface {
 }
 
 type manifestRetriever struct {
-	logger         log.FieldLogger
-	s3API          s3iface.S3API
-	bucket, prefix string
+	logger log.FieldLogger
+	s3API  s3iface.S3API
+	bucket string
+	prefix string
 }
 
-func NewManifestRetriever(logger log.FieldLogger, region, bucket, prefix string) ManifestRetriever {
-	awsSession := session.Must(session.NewSession())
-	client := s3.New(awsSession, aws.NewConfig().WithRegion(region))
+func NewManifestRetriever(logger log.FieldLogger, region, bucket, prefix, caBundlePath string) (ManifestRetriever, error) {
```

Comment:
```suggestion
func proxyAwareManifestRetriever(logger log.FieldLogger, region, bucket, prefix, caBundlePath string) (ManifestRetriever, error) {
```

Reply:
Or perhaps that's a bogus function name to suggest, but what's going on with `RetrieveManifests()` vs. `NewManifestRetreiver()` Is the New one getting new manifests or is it a new version of some other function?

Reply:
There's this `manifestRetriver` structure type that's contained in the `aws` package. The `NewManifestRetriever` is a function that spits out an initialized `manifestRetriever` object. The `RetrieveManifests(...)` is a _method_ on the `manifestRetriever` type that's responsible for retrieving these billing manifests from the path to the bucket that was provided.

We may want to investigate renaming these functions/methods for clarity. I don't have any great solutions off the top of my head, but renaming the `RetrieveManifests` method is probably the most straightforward.

Reply:
okay, sounds maybe out of scope for this PR then.